sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/KlaviyoApiRequest.kt

```diff
@@ -149,7 +80,12 @@ internal open class KlaviyoApiRequest(
             val time = json.getLong(TIME_JSON_KEY)
             val uuid = json.getString(UUID_JSON_KEY)
 
-            return KlaviyoApiRequest(urlPath, method, time, uuid).apply {
+            return when (json.optString(TYPE_JSON_KEY)) {
+                ProfileApiRequest::class.simpleName -> ProfileApiRequest(time, uuid)
+                EventApiRequest::class.simpleName -> EventApiRequest(time, uuid)
+                PushTokenApiRequest::class.simpleName -> PushTokenApiRequest(time, uuid)
```

Comment:
One other question I have, just to make sure my understanding is right:
We changed the constructors to each request object so that we can instantiate a barebones request object (cast to the correct type this time) and then the `fromJson` method will make sure that the jsonified fields (like `body` and `headers`) which are needed to send the request are re-populated. We have a second constructor taking in SDK-relavant information for instantiation when a user first queues a request which takes the given SDK params and translates them into the jsonified fields.
Correct?

Reply:
Exactly